components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2531,15 +2531,14 @@ private void overrideResidentIdpEPUrls(IdentityProvider residentIDP)
             throws IdentityProviderManagementException {
 
         // Not all endpoints are persisted. So we need to update only a few properties.
+        String samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true);
+        samlSSOUrl = resolveAbsoluteURL(IdPManagementConstants.SAMLSSO, samlSSOUrl);
```

Comment:
So which invokes this method and why that tenant param does not need to be applied here

Reply:
getResidentIdP invokes this overrideResidentIdpEPUrls(IdentityProvider residentIDP). 

So, eventually, FederationAuthenticationConfig properties for samlsso are overwritten through this.

If the query parameter is needed to be appended to the residentIdP configs for samlsso, we need to append that here.